### DISCUSSION TITLE
Solito Router - not added to the navigation history

### DISCUSSION THREAD

**Discussion by sunnychaganty** [nandorojo/solito#190] [Q&A]
**Use case:**

1. As a user navigate to a /dashboard router (either Web or Expo app)
2. Then navigate (Solito Router push()) to /search?q=something
3. Then navigate to a different /search?q=something else

**Expected:** 
when the back button is click (Web or Expo) - the user should do the previous screen (Step 2) & then when the back button is clicked again, go to the /dashboard

**Observed:** 
user goes to the /dashboard screen & misses the screen in step 2

I think the router is doing a client side state change and not adding to the router history. Due to this, if the router is routing to the same screen with different params, the interim screens will be skipped when the back button of browser or Expo app is clicked.

Is anyone else facing this issue with Solito Router?

**Comment by nandorojo** [comment]
i’m confused. step 2 and 3 are the exact same URL?

**Reply by sunnychaganty** [reply]
step 2: /search?types=cars - rending a list of car model
then user clicks on a model, so the query changes to:
step 3: /search/types=cars&model=bmw
now, the screen is showing a list of BMW's - but then since the Search Result screen is reused, its the same View with different params
in this case, if the use clicks on the back button, then instead of showing the /search?types=cars, the history is taking the user back to the /dashboard screen. Hope this helpss

**Reply by nandorojo** [reply]
it’s just using next router under the hood, so it should work on web. native won’t have the same behavior without something custom

**Reply by sunnychaganty** [reply]
So: for web + Native - it works fine if its routed to a different view - the history seems to be working fine for both platforms. 
This issue is only if we are routing to the same view with different URL params, so somehow the interim history seems to be lost.

**Comment by timoisalive** [comment]
We ran into the same issue. We have three tabs, and each tab is a stack navigator. In the stacks we render content which is defined mostly by the parameters instead of the screen name. For example there might be User screens, which link to other User screens, and those should be added to history and stacked with a transition on native. Think about navigating Instagram profiles, a stack which shows Profile screens defined by profile id.

Here's the feature in React Navigation docs: https://reactnavigation.org/docs/navigating/#navigate-to-a-route-multiple-times
Which makes me think if Solito router `push` should actually be `navigate` and `push` would be added to do exactly the above...

We have solved this by using `useNavigation` from `@react-navigation/native` or `expo-router` on native and `useRouter` from `solito` on web, but it would be nice not to have to resort into workarounds like this in such common navigational feature.

But anyhow, thanks for the (otherwise) awesome library! 🙏

**Reply by nandorojo** [reply]
It’s a bit confusing to understand what you want to happen vs. what is. Are you talking about recursive stack pushing within the same tab? 

https://github.com/nandorojo/solito/issues/174
https://github.com/nandorojo/solito/issues/110

**Reply by timoisalive** [reply]
Yes, recursive stack pushing within same tab.

I'd like to push new screens (eg. `/user/${randomId}`) into the stack and see them animating in, like normally in a stack. Current behavior is, that the user screen gets replaced by the new user screen (no animation), and if I go back, I don't end up in the previous user screen, but in the screen BEFORE the user screens.

#110 is an interesting one that I didn't even run into yet, but will very soon! Luckily seems like Evan's got something coming up for it already: https://github.com/expo/router/pull/148
